examples/external-data/src/mock-external-data-service/service.ts

```diff
@@ -142,8 +190,75 @@ export async function initializeExternalDataService(props: ServiceProps): Promis
 					`Registered for webhook notifications at URL: "${subscriberUrl}".`,
 				),
 			);
-			result.send();
+		} catch (error) {
+			if (error instanceof ApiError) {
+				console.warn(formatLogMessage(error.message));
+				result.status(error.code).json({ message: error.message });
+			} else {
+				console.error(error);
+				throw error;
+			}
 		}
+
+		result.send();
+	});
+
+	/**
+	 * Unregisters the specified URL from receiving notifications for the specified external task list id.
+	 *
+	 * Expected request body format: {@link UnregisterWebhookRequest}
+	 */
+	expressApp.post("/unregister-webhook", (request: UnregisterWebhookRequest, result) => {
+		try {
+			// 1. Validate request data
+			const subscriberUrl = request.body.url;
+			if (typeof subscriberUrl !== "string") {
+				throw new InvalidRequestError("Missing or unexpected data in request body");
+			} else if (isWebUri(subscriberUrl) === undefined) {
+				throw new InvalidRequestError(
+					`Provided subscriber URL is invalid ${subscriberUrl}`,
+				);
+			}
+			const externalTaskListId = request.body.externalTaskListId;
+			if (externalTaskListId === undefined || typeof externalTaskListId !== "string") {
+				throw new InvalidRequestError(
+					`Missing or malformed taskListId in request url: ${externalTaskListId}`,
+				);
+			}
+
+			// 2. Find cooresponding webook for the given externalTaskListId
+			const webhook = webhookCollection.get(externalTaskListId);
+			if (webhook === undefined) {
+				throw new InvalidRequestError(
+					"Provided externalTaskListId has no outstanding webhooks",
+				);
+			}
+
+			// 3. Webhook exists, attempt to remove the subscriber from the webhook
+			if (!webhook.subscribers.includes(subscriberUrl)) {
+				// 3a. Webhook exists but the provided subscriber is not subscribed with the webhook.
+				const resultMessage =
+					"Provided subscriberUrl does not have a webhook registered for the given externalTaskListId";
+				result.status(200).json({ message: resultMessage });
+				console.info(formatLogMessage(resultMessage));
```

Comment:
Should this come before we send the response? Only because that is the order we seem to be using elsewhere.

Reply:
Yes. Updated.

Reply:
I like it better like this but I think `result.status().json()` does not actually send anything, that only happens at `result.send()` below, right? Anyway, `.status()` or `.json()` could fail so I agree that logging to console first makes sense.